tests/main/lxd/task.yaml

```diff
@@ -86,6 +86,8 @@ execute: |
     lxd.lxc exec my-ubuntu -- su -c "/snap/bin/test-snapd-tools.echo from-the-inside" ubuntu | MATCH from-the-inside
     echo "And as root"
     lxd.lxc exec my-ubuntu -- test-snapd-tools.echo from-the-inside | MATCH from-the-inside
+    echo "We can also remove snaps successfully"
+    lxd.lxc exec my-ubuntu -- snap remove test-snapd-tools
```

Comment:
Hmm, unless I'm missing something, this doesn't actually test the fix, as installing & removing worked fine before in such cases too. To hit the mount ordering issue you need to install, reboot and then remove.

Reply:
Hmm, are you sure? I don't recall how reboot affects this issue. I think the issue was with _refreshes_ (where new things got mounted outside).